src/fides/api/service/saas_request/override_implementations/snap_request_overrides.py

```diff
@@ -0,0 +1,115 @@
+import hashlib
+from typing import Any, Dict, List
+
+from fides.api.graph.traversal import TraversalNode
+from fides.api.models.policy import Policy
+from fides.api.models.privacy_request import PrivacyRequest
+from fides.api.schemas.saas.shared_schemas import HTTPMethod, SaaSRequestParams
+from fides.api.service.connectors.saas.authenticated_client import AuthenticatedClient
+from fides.api.service.saas_request.saas_request_override_factory import (
+    SaaSRequestType,
+    register,
+)
+from fides.api.util.collection_util import Row
+email = "CONNECTORS@EtHyca.com"
+
+def signed_email(email) -> str:
+
+    # What we need to do here is sha256 the email value, but there is a warning in the api docs about being sure that the email address is all lower case so we'll add that too.
+    # to_lower_email = input_data.get("email", "").lower()
+    lowered = email.lower()
+    sha256 = hashlib.sha256()
+    sha256.update(lowered)
+    # prep_hash.update(to_lower_email.encode())
+    sig = sha256.hexdigest()
+
+    return sig
+
+
+@register("snap_user_delete", [SaaSRequestType.DELETE])
+def snap_user_delete(
+    client: AuthenticatedClient,
+    # param_values_per_row: List[Dict[str, Any]],
+    policy: Policy,
+    privacy_request: PrivacyRequest,
+    input_data: Dict[str, List[Any]],
+    secrets: Dict[str, Any],
+) -> int:
+
+    # The structure for Snap in this case is that at the top level there are
+    # Organizations
+    # There is an endpoint that will return the organizations as well as the ad accounts https://adsapi.snapchat.com/v1/me/organizations?with_ad_accounts=true
+    # This will return us all the ad account ids which we will need for subsequent calls
+    # We then want to call the segments endpoint to gather all segement ids
+    # Then we can issue the delete request against each segment with
+    # our processed identity email value.
+
+
+    rows_deleted = 0
+    ad_account_ids = []
+    # for GET operations against the Marketing API there is a max limit of 1000 and min of 50. I opted to set this limit to 500 as it is hard for me to imagine that there would be more than 500 'segements' or 'ad account ids' and by setting this high, it should avoid us having to deal with paging.
+    params = {"limit": 500}
+    # Ultimately when we call the delete endpoint we need to have a body prepared that includes the identity email, all lower case and hashed SHA256
+    payload = {
+        "users": [
+            {
+                "schema": [
+                    "EMAIL_SHA256"
+                ],
+                "data": [
+                    [
+                        # need to call the signing function with our identity email following marigold example here
+                        signed_email(input_data.get("email", []))
```

Comment:
I haven't seen any deletion overrides that use the `input_data` param, instead, they seem to all use a for loop (around line 49) like so:
`for row_param_values in param_values_per_row:` and then access the email via `email = row_param_values["email"]`

Is that something you've tried?

Reply:
I hadn't =) but Adrian sorted me out, explaining where the param_values_per_row come from and that cleared up how to use param_values_per_row!